File: examples/languages/sample.go
// Go sample demonstrating syntax highlighting
package main

import (
	"fmt"
	"math"
)

// Color represents an RGB color
type Color struct {
	R, G, B uint8
}

// NewColor creates a new Color with clamped values
func NewColor(r, g, b int) Color {
	clamp := func(v int) uint8 {
		if v < 0 {
			return 0
		}
		if v > 255 {
			return 255
		}
		return uint8(v)
	}
	return Color{clamp(r), clamp(g), clamp(b)}
}

// ToHex converts the color to a hex string
func (c Color) ToHex() string {
	return fmt.Sprintf("#%02x%02x%02x", c.R, c.G, c.B)
}

// Lighten returns a lightened version of the color
func (c Color) Lighten(amount float64) Color {
	adjust := func(v uint8) uint8 {
		result := float64(v) + (255.0-float64(v))*amount
		return uint8(math.Min(255, result))
	}
	return Color{adjust(c.R), adjust(c.G), adjust(c.B)}
}

// GeneratePalette creates a palette of colors
func GeneratePalette(base Color, count int) []Color {
	colors := make([]Color, count)
	colors[0] = base

	for i := 1; i < count; i++ {
		angle := (360.0 / float64(count)) * float64(i)
		rad := angle * math.Pi / 180.0
		colors[i] = NewColor(
			int(math.Abs(math.Cos(rad))*255),
			int(math.Abs(math.Sin(rad))*255),
			int(base.B),
		)
	}

	return colors
}

func main() {
	base := NewColor(255, 128, 0)
	fmt.Printf("Base color: %s\n", base.ToHex())

	palette := GeneratePalette(base, 5)

	for i, color := range palette {
		lightened := color.Lighten(0.2)
		fmt.Printf("%d: %s -> %s\n", i, color.ToHex(), lightened.ToHex())
	}
}

File: examples/languages/sample.py
#!/usr/bin/env python3
"""Python sample demonstrating syntax highlighting"""

from typing import List, Tuple
import math


class Color:
    """Represents an RGB color"""

    def __init__(self, r: int, g: int, b: int):
        self.r = max(0, min(255, r))
        self.g = max(0, min(255, g))
        self.b = max(0, min(255, b))

    def to_hex(self) -> str:
        """Convert color to hex string"""
        return f"#{self.r:02x}{self.g:02x}{self.b:02x}"

    def lighten(self, amount: float = 0.1) -> 'Color':
        """Return a lightened version of this color"""
        adjust = lambda c: min(255, int(c + (255 - c) * amount))
        return Color(adjust(self.r), adjust(self.g), adjust(self.b))

    def __repr__(self) -> str:
        return f"Color({self.r}, {self.g}, {self.b})"


def generate_palette(base: Color, count: int) -> List[Color]:
    """Generate a palette by varying the hue"""
    colors = [base]
    for i in range(1, count):
        # Simple hue rotation
        angle = (360 / count) * i
        rad = math.radians(angle)
        # This is a simplified color rotation
        colors.append(Color(
            int(abs(math.cos(rad)) * 255),
            int(abs(math.sin(rad)) * 255),
            base.b
        ))
    return colors


def main():
    # Create base color
    base = Color(255, 128, 0)
    print(f"Base color: {base.to_hex()}")

    # Generate palette
    palette = generate_palette(base, 5)

    # Display results
    for i, color in enumerate(palette):
        lightened = color.lighten(0.2)
        print(f"{i}: {color.to_hex()} -> {lightened.to_hex()}")


if __name__ == "__main__":
    main()
